### ISSUE TITLE
TransactionValidationError. Tx with such id aready present.

### ISSUE THREAD

**Issue by dbykov** [wavesplatform/Waves#564]
I'm using waves_0.7.8_all.deb on Ubuntu. After reboot, node gives me this error:

```
java.lang.Exception: TransactionValidationError(GenericError(Tx id(exc. for some PaymentTransactions) cannot be duplicated. Current height is: 633381. Tx with such id aready present),{"type":4,"id":"EHUNme71H6LME1Whz7jg8Ajs7XebUuZEhdD9LsgHX8iF","sender":"3PJuaHc38E2GNaU5t2dHupRXAGT54DMMhCk","senderPublicKey":"4Ki6tWAF8jxHfZqThQseEa71pB5VYEJAEyey7U5zMiiJ","fee":100000,"timestamp":1503316280387,"signature":"3huCjogZpxmGpXhQ6HXU8Fq6shNQ8iRKcQeAamSKhR4hEfXnKQtLGZLDn9j4Njb6PJjQB1fUFxkfn3ddarsrTsV6","recipient":"3P9D1oU27JVxf3i9y2zWhh2fBGkc4fovGEo","assetId":"4eT6R8R2XuTcBuTHiXVQsh2dN2mg3c2Qnp95EWBNHygg","amount":1288984270,"feeAsset":null,"attachment":""})
```

How can i fix it?

**Comment by dbykov** [comment]
Full log:

```systemd[1]: Started Waves node.
waves[10840]: INFO  c.w.Application$ - Starting...
waves[10840]: INFO  kamon.Kamon$Instance - Initializing Kamon...
waves[10840]: INFO  a.event.slf4j.Slf4jLogger - Slf4jLogger started
waves[10840]: INFO  c.w.Application$ - Waves v0.7.8 Blockchain Id: W
waves[10840]: INFO  c.w.s.BlockchainUpdaterImpl - Constructing BlockchainUpdaterImpl, total blocks: 633600, persisted: 633380, topMemDiff: 0, bottomMemDiff: 0
waves[10840]: INFO  c.w.s.BlockchainUpdaterImpl - State rebuild started, total blocks: 633600, persisted: 633380, topMemDiff: 0, bottomMemDiff: 0
waves[10840]: ERROR c.w.actor.RootActorSystem$ - Error while initializing actor system wavesplatform
waves[10840]: java.lang.Exception: TransactionValidationError(GenericError(Tx id(exc. for some PaymentTransactions) cannot be duplicated. Current height is: 633381. Tx with such id aready present),{"type":4,"id":"EHUNme71H6LME1Whz7jg8Ajs7XebUuZEhdD9LsgHX8iF","sender":"3PJuaHc38E2GNaU5t2dHupRXAGT54DMMhCk","senderPublicKey":"4Ki6tWAF8jxHfZqThQseEa71pB5VYEJAEyey7U5zMiiJ","fee":100000,"timestamp":1503316280387,"signature":"3huCjogZpxmGpXhQ6HXU8Fq6shNQ8iRKcQeAamSKhR4hEfXnKQtLGZLDn9j4Njb6PJjQB1fUFxkfn3ddarsrTsV6","recipient":"3P9D1oU27JVxf3i9y2zWhh2fBGkc4fovGEo","assetId":"4eT6R8R2XuTcBuTHiXVQsh2dN2mg3c2Qnp95EWBNHygg","amount":1288984270,"feeAsset":null,"attachment":""})
systemd[1]: waves.service: Main process exited, code=exited, status=1/FAILURE
systemd[1]: waves.service: Unit entered failed state.
systemd[1]: waves.service: Failed with result 'exit-code'.
systemd[1]: Stopped Waves node.
```

**Comment by MeInTheFuture** [comment]
I had the same problem with my waves node:
* how I fixed it: by deleting the state.dat file and waiting for like 17h for the "repair" by waves for the 765456 blocks

<br>

more info given [here](https://steemit.com/waves/@thefuturenow/my-waves-node-did-not-start-anymore-and-how-i-fixed-this)

<br>

in my case it seems due to less RAM waves could not finish some tasks and this rendered waves "unusable" afterwards :-(

<br>
it would be great if waves could "heal itself" without user intervention in such cases



---
* waves 0.7.9
* java 1.8.0_151

on request more info available

**Comment by alexeykiselev** [comment]
Solved since version 0.8. Lack of memory is not the issue since 0.9 with LevelDB.
But it's hard to repair the corrupted state anyway. The simplest solution for such situations is to rebuild the state from scratch.

**Comment by Shashwat78** [comment]
[info] 2018-02-16 13:12:27,901 ERROR [main] c.w.actor.RootActorSystem$ - Error while initializing actor system wavesplatform
[info] java.util.NoSuchElementException: Either.right.get on Left
[info] 	at scala.util.Either$RightProjection.get(Either.scala:641)
[info] 	at scorex.block.Block$.$anonfun$genesisTransactions$1(Block.scala:269)
[info] 	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
[info] 	at scala.collection.Iterator.foreach(Iterator.scala:929)
[info] 	at scala.collection.Iterator.foreach$(Iterator.scala:929)
[info] 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
[info] 	at scala.collection.IterableLike.foreach(IterableLike.scala:71)
[info] 	at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
[info] 	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
[info] 	at scala.collection.TraversableLike.map(TraversableLike.scala:234)
[info] 	at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
[info] 	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
[info] 	at scorex.block.Block$.genesisTransactions(Block.scala:268)
[info] 	at scorex.block.Block$.genesis(Block.scala:278)
[info] 	at com.wavesplatform.package$.checkGenesis(package.scala:11)
[info] 	at com.wavesplatform.Application.run(Application.scala:102)
[info] 	at com.wavesplatform.Application$.$anonfun$main$3(Application.scala:369)
[info] 	at com.wavesplatform.Application$.$anonfun$main$3$adapted(Application.scala:343)
[info] 	at com.wavesplatform.actor.RootActorSystem$.start(RootActorSystem.scala:25)
[info] 	at com.wavesplatform.Application$.main(Application.scala:343)
[info] 	at com.wavesplatform.Application.main(Application.scala)
[error] java.lang.RuntimeException: Nonzero exit code returned from runner: 1
[error] 	at sbt.ForkRun.processExitCode$1(Run.scala:33)
[error] 	at sbt.ForkRun.run(Run.scala:42)
[error] 	at sbt.Defaults$.$anonfun$bgRunTask$5(Defaults.scala:1168)
[error] 	at sbt.Defaults$.$anonfun$bgRunTask$5$adapted(Defaults.scala:1163)
[error] 	at sbt.internal.BackgroundThreadPool.$anonfun$run$1(DefaultBackgroundJobService.scala:366)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
[error] 	at scala.util.Try$.apply(Try.scala:209)
[error] 	at sbt.internal.BackgroundThreadPool$BackgroundRunnable.run(DefaultBackgroundJobService.scala:289)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] 	at java.lang.Thread.run(Thread.java:748)
[error] (Compile / run) Nonzero exit code returned from runner: 1
  

any help please ??
